src/Mollie/WC/ApplePayButton/AjaxRequests.php

```diff
@@ -76,9 +76,13 @@ public function validateMerchant()
             return;
         }
         $validationUrl = $applePayRequestDataObject->validationUrl;
-        $completeDomain = get_site_url();
-        $removeHttp = ["https://", "http://", "/"];
-        $domain = str_replace($removeHttp, "", $completeDomain);
+        $completeDomain = get_site_url(); //https://www.example.com/bla/bla
+        $removeHttp = ["https://", "http://"];
+        $regex = '/.+\.\w+\/?((\w*\/*)*)/i';//captures in $1 strings with the form bla or bla/ or bla/bla
+        $domain = str_replace($removeHttp, "", $completeDomain);//www.example.com/bla/bla
+        $ending = preg_replace($regex, '$1', $domain);
+        $domain = str_replace($ending, "", $domain);//www.example.com/
+        $domain = str_replace("/", "", $domain);//www.example.com
```

Comment:
Although it works without any problems, you may find it convenient to use the [ parse_url()](https://www.php.net/manual/en/function.parse-url.php) function.

So, it could be like 
```php
$domain = parse_url(get_site_url(), PHP_URL_HOST);
```